static/src/javascripts/projects/common/modules/identity/email-prefs.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import reqwest from 'reqwest';
+import fastdom from 'lib/fastdom-promise';
+import loadEnhancers from './modules/loadEnhancers';
+
+import { push as pushError } from './modules/show-errors';
+import { addUpdatingState, removeUpdatingState } from './modules/button';
+import {
+    getCsrfTokenFromElement,
+    getNewsletterHtmlPreferenceFromElement,
+} from './modules/fetchFormFields';
+
+const submitNewsletterHtmlPreference = (
+    csrfToken: string,
+    htmlPreference: string
+): Promise<void> => {
+    const formData = new FormData();
+    formData.append('csrfToken', csrfToken);
+    formData.append('htmlPreference', htmlPreference);
+
+    return reqwest({
+        url: '/email-prefs',
+        method: 'POST',
+        data: formData,
+        processData: false,
+    });
+};
+
+const bindHtmlPreferenceChange = (buttonEl: HTMLButtonElement): void => {
+    buttonEl.addEventListener('click', () =>
+        Promise.all([
+            getCsrfTokenFromElement(buttonEl),
+            getNewsletterHtmlPreferenceFromElement(buttonEl),
+            addUpdatingState(buttonEl),
+        ])
+            .then(([csrfToken: string, htmlPreference: string]) =>
+                submitNewsletterHtmlPreference(csrfToken, htmlPreference)
+            )
+            .catch((err: Error) => {
+                pushError(err, 'reload').then(() => {
+                    window.scrollTo(0, 0);
+                });
+            })
+            .then(() => {
+                removeUpdatingState(buttonEl);
+            })
+    );
+};
+
+const bindModalCloser = (buttonEl: HTMLElement): void => {
+    buttonEl.addEventListener('click', () => {
+        const modalEl: ?Element = buttonEl.closest('.manage-account__modal');
+        if (modalEl) {
+            modalEl.classList.remove('manage-account__modal--active');
+        }
+    });
+};
+
+const toggleFormatModal = (buttonEl: HTMLElement): void => {
```

Comment:
Naming between `bindModalCloser` and `toggleFormatModal` is a bit inconsistent. What do you think about:

- modalCloserBind
- modalFormatToggle

?

Reply:
agreed. these were a bit messy because they sort of were placeholders for a more comprehensive modal api that didn't end up happening